consensus/src/checkpoint.rs

```diff
@@ -112,47 +117,221 @@ mod tests {
+            Some(blockchain_length_relativity) => {
+                a.body.consensus_state.curr_global_slot.slots_per_epoch.0 - 4
+            } // -1 to bring into inclusive range, -3 to provide 2 slots of fudge room
+            None => a.body.consensus_state.curr_global_slot.slots_per_epoch.0 - 1, // -1 to bring into inclusive range
+        };
+
+        let min_a_curr_epoch_slot_defaut = 0;
+        let min_a_curr_epoch_slot_sum = min_a_curr_epoch_slot_defaut + min_a_curr_epoch_slot;
+        let slot = thread_rng().gen_range(min_a_curr_epoch_slot_sum..max_epoch_slot);
+        let length = gen_num_blocks_in_slots(
+            default_slot_fill_rate,
+            default_slot_fill_rate_delta,
+            slot as f64,
+        );
+
+        a.body.consensus_state.curr_global_slot.slot_number.0 = slot;
+        a.body.consensus_state.blockchain_length.0 = length as u32;
+
+        // Randomized root_epoch_position for more robust test
+        let root_epoch_position = base_root_epoch_position;
+        let (_, root_epoch_length) = root_epoch_position;
+
+        let length_till_curr_epoch = root_epoch_length
+            + a.body.consensus_state.staking_epoch_data.epoch_length.0
+            + a.body.consensus_state.next_epoch_data.epoch_length.0;
+        let a_curr_epoch_length = length_till_curr_epoch;
+
+        // Handle relativity constriants for second state.
+        let a_curr_epoch_slot = &a.body.consensus_state.curr_global_slot.slot_number;
+
+        // Generate second state position by extending the first state's position
+        let protocol_constants = ProtocolConstantsTemp::default();
+        let max_epoch_slot = protocol_constants.slots_per_epoch.0 - 1;
+
+        // This invariant needs to be held for the position of `a`
+        assert!(max_epoch_slot > a_curr_epoch_slot.0 + 2);
+
+        // Assume there is a next block in the slot directly preceeding the block for `a`
+        let added_slots = thread_rng().gen_range(a_curr_epoch_slot.0 + 2..max_epoch_slot);
+
+        let added_blocks = gen_num_blocks_in_slots(
+            default_slot_fill_rate,
+            default_slot_fill_rate_delta,
+            added_slots as f64,
+        );
+
+        b.body.consensus_state.curr_global_slot.slot_number.0 =
+            a_curr_epoch_slot.0 + added_slots + 1;
+        b.body.consensus_state.blockchain_length.0 = a_curr_epoch_length + added_blocks as u32 + 1;
+    }
+
     #[test]
-    fn test_is_short_range() {
+    #[wasm_bindgen_test]
+    fn equal_state_in_short_fork_range() {
         let mut genesis: ProtocolState = Default::default();
-        init_checkpoints(&mut genesis);
+        init_checkpoints(&mut genesis).unwrap();
+
+        let mut a: ProtocolState = Default::default();
+        let mut b: ProtocolState = Default::default();
+
+        gen_spot(&mut a);
+        gen_spot(&mut b);
+
         let mut c0: ProtocolStateChain = ProtocolStateChain(vec![]);
```

Comment:
let c0: ProtocolStateChain = ProtocolStateChain(vec![a]);

Reply:
Then we don't need to use push here?

Reply:
fixed